pkg/core/account/accounts.go

```diff
@@ -125,9 +118,54 @@ func (a *Accounts) ForEach(callback func(id iotago.AccountID, pool *Pool) bool)
 	a.accountPools.ForEach(callback)
 }
 
-// SeatedAccounts creates a new SeatedAccounts instance, that maintains the seats of the given members.
-func (a *Accounts) SeatedAccounts(members ...iotago.AccountID) *SeatedAccounts {
-	return NewSeatedAccounts(a, members...)
+// SeatedAccounts creates a new SeatedAccounts instance,
+// that maintains the seat indices of re-elected members from the previous committee, if provided.
+func (a *Accounts) SeatedAccounts(optPrevCommittee ...*SeatedAccounts) *SeatedAccounts {
+	// If optPrevCommittee not given, use empty SeatedAccounts instance as default to skip the behavior to handle re-elected members.
+	prevCommittee := NewSeatedAccounts(NewAccounts())
+	if len(optPrevCommittee) > 0 {
+		prevCommittee = optPrevCommittee[0]
+	}
+
+	newCommittee := NewSeatedAccounts(a)
+
+	// If previous committee exists the seats need to be assigned and re-elected committee members must retain their SeatIndex.
+	// Keep track of re-elected members' seat indices.
+	takenSeatIndices := ds.NewSet[SeatIndex]()
+
+	// Assign re-elected members' SeatIndex.
+	a.ForEach(func(id iotago.AccountID, _ *Pool) bool {
+		if seatIndex, seatExists := prevCommittee.GetSeat(id); seatExists {
+			newCommittee.Set(seatIndex, id)
+			takenSeatIndices.Add(seatIndex)
```

Comment:
How exactly does this behave if `prevCommittee` is bigger than `newCommittee`? In the test the committees are equal size iiuc. Would be good to have a test case for that.

Reply:
I extended the test to cover going to a smaller and bigger committee.